src/container.c

```diff
@@ -224,6 +224,55 @@ static int container_setup_mount(struct hyper_container *container)
 	return 0;
 }
 
+static int container_recreate_file(char *filename)
+{
+	fprintf(stdout, "recreate file %s\n", filename);
+	hyper_unlink(filename);
+	return hyper_create(filename);
+}
+
```

Comment:
The init layer is used by docker to over ride those files/dirs in the image, from container layer point of view. If performance is a concern here, yes we can skip unlink in some cases, such as those files exist and are empty, providing type and mode match.

Actually, docker would simply unlink, then stat and only create if stat fails with ENOENT. So to have similar behaviour and minimal performance impact, we can do stat first and skip unlink/create if file size is 0, otherwise we do unlink + create.